gqcp/include/Basis/Transformations/SimpleOrbitalRotationGenerators.hpp

```diff
@@ -0,0 +1,148 @@
+// This file is part of GQCG-GQCP.
+//
+// Copyright (C) 2017-2020  the GQCG developers
+//
+// GQCG-GQCP is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// GQCG-GQCP is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-GQCP.  If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+
+
+#include "Basis/SpinorBasis/OccupationType.hpp"
+#include "Basis/SpinorBasis/OrbitalSpace.hpp"
+#include "Mathematical/Representation/SquareMatrix.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ *  MARK: SimpleOrbitalRotationGenerator implementation
+ */
+
+/**
+ *  A set of orbital rotation generators that can be represented by a single vector.
+ *   
+ *  This class is used as a base class for `ROrbitalRotationGenerator` and `GOrbitalRotationGenerator`, since they are both expressed using a single vector of kappa_PQ values, as opposed to `UOrbitalRotationGenerator`, which uses separate vectors for alpha- and beta- generators. The word 'simple' is used here as an antonym for 'compound'.
+ * 
+ *  @tparam _Scalar                                           The scalar type used for an orbital rotation generator: real or complex.
+ *  @tparam _DerivedOrbitalRotationGenerator                  The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+ */
+template <typename _Scalar, typename _DerivedOrbitalRotationGenerators>
+class SimpleOrbitalRotationGenerators {
+
+public:
+    // The scalar type used for an orbital rotation generator: real or complex.
+    using Scalar = _Scalar;
+
+    // The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+    using DerivedOrbitalRotationGenerators = _DerivedOrbitalRotationGenerators;
+
+private:
+    // The number of orbitals (spinors for the general(ized) case, spin-orbitals for the restricted and unrestricted case) that can be rotated using these orbital rotation generators.
+    size_t number_of_orbitals;
+
+    // The strict lower triangle of the kappa matrix.
+    VectorX<Scalar> v;
```

Comment:
Could you add to the documentation in which 'major' mode this vector holds the entries of the matrix? If I'm not mistaken, I think it's in _column major_ format.